Client/Frontend/Browser/SearchLoader.swift

```diff
@@ -33,12 +37,21 @@ class _SearchLoader<UnusedA, UnusedB>: Loader<Cursor<Site>, SearchViewController
                 return
             }
 
-            self.history.getSitesByFrecencyWithLimit(100, whereURLContains: query).uponQueue(dispatch_get_main_queue()) { result in
+            if let inProgress = inProgress {
+                inProgress.cancel()
```

Comment:
If we can do that, yeah. Trying to avoid running DB code on the main thread by accident.